### PR TITLE
packages.json is not needed for detection

### PR DESCRIPTION
**Pull request by bleach**
## What

Remove a line that stated that packages.json is required for detection of the
correct buildpack.
## Why

The current wording is inaccurate and might mean that users form an incorrect mental model of how Cloud Foundry works.
## How to review

There's no change to to the procedure itself, so the review process should be someone independently reviewing the [nodejs buildpack](https://docs.cloudfoundry.org/buildpacks/node/index.html) and [buildpack detection docs](https://docs.cloudfoundry.org/buildpacks/detection.html) and making sure my understanding of the behaviour is correct
## Who can review

Anyone but @bleach

### CODE REVIEW

File: docs/deploying_apps/deploying_node_js.md [hunk 1]
```diff
@@ -19,8 +19,6 @@ These steps assume you have already carried out the setup process explained in t
 
 3. Include an npm ``package.json`` file to specify dependencies. The file should also specify a `start` command used to launch the app.
   
-    The presence of this file is essential to have the application detected as Node.js.
```
Comment (mtekel) [inline docs/deploying_apps/deploying_node_js.md:22]: If you look in https://github.com/cloudfoundry/nodejs-buildpack/blob/master/bin/detect - it really does check for package.json presence - as only thing to detect app is node.js

The confusion for me is in `the file should also specify start command` - they mean app manifest, not package.json. This should be probably written more cleanly.

Comment (bleach) [inline docs/deploying_apps/deploying_node_js.md:22]: I agree that it requires it and the script is called `detect.yml`, but not to detect which buildpack to use. We've already told the user to add this to `manifest.yml`

```
  buildpack: nodejs_buildpack
```

I'll reword this to say the buildpack always needs it.

Comment (benh-gds) [inline docs/deploying_apps/deploying_node_js.md:22]: @mtekel Just want to clarify about the start command: it's my understanding that you can put it in either package.json (using "start:") or manifest.yml (using command:). The original docs said to put it in package.json, because  the ability to put it in the manifest  may be removed, and they cite this comment from 2014: https://github.com/cloudfoundry/nodejs-buildpack/pull/11#issuecomment-67666273 

However, it doesn't seem like the option to put it in the manifest has been removed - so should we now recommend that people do that?
